app/assets/javascripts/govuk_publishing_components/components/checkboxes.js

```diff
@@ -5,90 +5,118 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}
+
+    this.applyAriaControlsAttributes(this.$module)
+
+    for (var i = 0; i < this.$checkboxes.length; i++) {
+      this.$checkboxes[i].addEventListener('change', this.handleCheckboxChange)
+    }
+
+    for (i = 0; i < this.$nestedCheckboxes.length; i++) {
+      this.$nestedCheckboxes[i].addEventListener('change', this.handleNestedCheckboxChange.bind(this))
+    }
+
+    for (i = 0; i < this.$exclusiveCheckboxes.length; i++) {
+      this.$exclusiveCheckboxes[i].addEventListener('change', this.handleExclusiveCheckboxChange)
+    }
+  }
+
+  GovukCheckboxes.prototype.handleCheckboxChange = function (event) {
+    if (window.GOVUK.analytics && window.GOVUK.analytics.trackEvent) {
+      // where checkboxes are manipulated externally in finders, suppressAnalytics
+      // is passed to prevent duplicate GA events
+      if (!event.detail || (event.detail && event.detail.suppressAnalytics !== true)) {
+        var $checkbox = event.target
+        var category = $checkbox.getAttribute('data-track-category')
+        if (category) {
+          var uncheckTrackCategory = $checkbox.getAttribute('data-uncheck-track-category')
+          if (!$checkbox.checked && uncheckTrackCategory) {
+            category = uncheckTrackCategory
           }
+          var action = $checkbox.getAttribute('data-track-action')
+          var options = $checkbox.getAttribute('data-track-options')
+          if (options) {
+            options = JSON.parse(options)
+          } else {
+            options = {}
+          }
+          options.value = $checkbox.getAttribute('data-track-value')
+          options.label = $checkbox.getAttribute('data-track-label')
+          window.GOVUK.analytics.trackEvent(category, action, options)
         }
-      })
-
-      $(scope).on('change', '[data-exclusive=true] input[type=checkbox]', function (e) {
-        var currentCheckbox = e.target
-        var checkboxes = currentCheckbox.closest('.govuk-checkboxes')
-        var exclusiveOption = $(checkboxes).find('input[type=checkbox][data-exclusive]')
-        var nonExclusiveOptions = $(checkboxes).find('input[type=checkbox]:not([data-exclusive])')
-
-        if (currentCheckbox.dataset.exclusive === 'true' && currentCheckbox.checked === true) {
-          nonExclusiveOptions.each(function () {
-            $(this).prop('checked', false)
-          })
-        } else if (currentCheckbox.dataset.exclusive !== 'true' && currentCheckbox.checked === true) {
-          exclusiveOption.prop('checked', false)
-        }
-      })
+      }
     }
+  }
 
-    this.toggleNestedCheckboxes = function (scope, checkbox) {
-      if (checkbox.checked) {
-        scope.find('input[type=checkbox]').prop('checked', true)
-      } else {
-        scope.find('input[type=checkbox]').prop('checked', false)
+  GovukCheckboxes.prototype.handleNestedCheckboxChange = function (event) {
+    var $checkbox = event.target
+    var $isNested = $checkbox.closest('.govuk-checkboxes--nested')
```

Comment:
Is it okay that `.closest()` isn't supported in Internet Explorer 11?

Reply:
I assumed it was required by `checkboxes.js` from `govuk-frontend`, but doesn't seem to be the case. It worked fine in my tests because the polyfill is required by other scripts (I think `error-summary`) and the component guide loads them all. I've explicitly required it in this script now. One of the downsides of rolling out individual component imports before setting up a build and tree-shaking mechanism is that we're going to serve it with every component that needs it.